src/util/create-addons-pages.js

```diff
@@ -0,0 +1,146 @@
+
+const addonDetail = `
+  id: name
+  name
+  displayName
+  description
+  icon
+  authors {
+    id: username
+    avatarUrl: gravatarUrl
+    name: username
+  }
+  weeklyDownloads
+  appearance: verified
+  verifiedCreator`;
+
+module.exports = function createAddonsPages({ actions, graphql }) {
+  const { createPage } = actions;
+  return graphql(
+    `
+      {
+        addons {
+          addonPages: top(sort: monthlyDownloads) {
+            ${addonDetail}
+            tags {
+              name
+              displayName
+              description
+              icon
+            }
+            compatibility {
+              name
+              displayName
+              icon
+            }
+            status
+            readme
+            publishedAt
+            repositoryUrl
+            homepageUrl
+          }
+          categoryPages: tags(isCategory: true) {
+            name
+            displayName
+            description
+            icon
+            addons: top(sort: monthlyDownloads) {
+              ${addonDetail}
+            }
+          }
+        }
+      }
+    `
+  )
+    .then(
+      ({
+        data: {
+          addons: { addonPages, categoryPages },
+        },
+      }) => {
+        addonPages.forEach((addon) => {
+          createPage({
+            path: `/addons/${addon.name}`,
+            component: path.resolve(
+              `./src/components/screens/AddonsDetailScreen/AddonsDetailScreen.js`
+            ),
+            context: {
+              ...addon,
+              tags: buildTagLinks(addon.tags),
+              readme: processor.processSync(addon.readme).toString(),
+            },
+          });
+        });
+
+        categoryPages.forEach((category) => {
+          createPage({
+            path: `/addons/${category.name}`,
+            component: path.resolve(
+              `./src/components/screens/AddonsCategoryScreen/AddonsCategoryScreen.js`
+            ),
+            context: {
+              category: category.displayName,
+              description: category.description,
+              addons: category.addons,
+            },
+          });
+        });
+      }
+    )
+    .then(() => fetchTagPages(createPage, graphql));
+};
+
+function fetchTagPages(createPage, graphql, skip = 0) {
+  return graphql(
+    `{
+      addons {
+        tagPages: tags(isCategory: false, limit: 30, skip: ${skip}) {
+          name
+          displayName
+          description
+          icon
+          relatedTags {
+            name
+            displayName
+            icon
+          }
+          addons: top(sort: monthlyDownloads) {
+            ${addonDetail}
+          }
+        }
+      }
+    }`
+  ).then(
+    ({
+      data: {
+        addons: { tagPages },
+      },
+    }) => {
+      if (tagPages && tagPages.length > 0) {
+        createTagPages(createPage, tagPages);
+        return fetchTagPages(createPage, graphql, skip + tagPages.length);
```

Comment:
Can we put a 100ms sleep between successive queries to give the connections a chance to close? It looks like I'm still getting warnings on the backend every time you build.